src/java/voldemort/common/nio/SelectorManagerWorker.java

```diff
@@ -122,12 +120,11 @@ else if(!selectionKey.isValid())
             reportException(e);
             close();
         } catch(IOException e) {
-            logger.info("Connection reset from " + socketChannel.socket() + " with message - "
-                        + e.getMessage());
+            logger.info("Connection reset from " + getDebugInfo() + " with message - " + e.getMessage());
```

Comment:
This message is misleading.  The entire request handling logic is run in overloads of the `read()` method, so connection reset i.e. the Unix `ECONNRESET` error, is only one of many possible reasons `IOException` might escape out here.  The message should say something like "IO exception from " ...